### PR TITLE
feat: Switch merchant API Recoil Addition

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantListHook.res
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantListHook.res:1]: Can we move this file under hooks ?

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantListHook.res:1]: It's related to Switch Merchant I guess it should be inside switchMerchant folder. Not as it's a generic hook

**Review by Riddhiagrawal001 — APPROVED** [review]

File: src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantUtils.res [hunk 1]
```diff
@@ -0,0 +1,31 @@
+type switchMerchantListResponse = {
+  merchant_id: string,
+  merchant_name: string,
+  mutable is_active: bool,
```
Comment (vsrivatsa-juspay) [inline src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantUtils.res:4]:
```suggestion
  is_active: bool,
```

Comment (vsrivatsa-juspay) [reply src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantUtils.res:4]: Not seeing the mutability used anywhere 🤔

File: src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantListHook.res [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  let url = getURL(~entityName=USERS, ~userType=#SWITCH_MERCHANT, ~methodType=Get, ())
+
+  async _ => {
+    try {
+      let res = await fetchDetails(url)
+      let typedValueOfResponse = res->SwitchMerchantUtils.convertListResponseToTypedResponse
+      setSwitchMerchantListAtom(._ => typedValueOfResponse)
+    } catch {
+    | Exn.Error(e) => {
+        let err = Exn.message(e)->Option.getOr("Failed to Fetch!")
```
Comment (vsrivatsa-juspay) [inline src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantListHook.res:14]: Should the exception be more specific?
Ignore this suggestion if you'd like to push logs here instead (in a later PR)

Comment (vsrivatsa-juspay) [reply src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantListHook.res:14]: We would want to have analytics over exceptions to identify and mitigate issues

**Review by vsrivatsa-juspay — APPROVED** [review]

**Pull request merged**